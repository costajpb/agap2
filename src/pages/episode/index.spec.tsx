import { findByRole, findByTestId, render } from "@testing-library/react"
import Episode from "."
import data from '../../domain/tv-show/repository/__fixtures__/tvshow.json'
import Entity from "../../domain/episode/entity"

const rawEpisode = data._embedded.episodes[0]

describe('pages/episode', () => {
    it('should show title, summary and cover image', async () => {
        const episode: Entity = {
            id: rawEpisode.id,
            title: rawEpisode.name,
            summary: rawEpisode.summary,
            coverImage: rawEpisode.image.original
        }

        const { container } = render(<Episode details={episode} />)

        expect((await findByRole(container, 'heading')).textContent).toBe(episode.title)
        expect((await findByTestId(container, 'summary')).textContent != '').toBeTruthy()
        expect((await findByRole(container, 'img')).getAttribute('src')).toBe(episode.coverImage)
    })
})