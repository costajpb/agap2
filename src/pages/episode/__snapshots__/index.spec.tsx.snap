// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/episode should render with a message when summary is not available 1`] = `
<div>
  <article
    class="sc-dcJsrY gxNnMq"
  >
    <h1
      class="sc-cWSHoV dKuyNJ"
    >
      <span
        class="sc-kMkxaj jWvPnq"
      >
        Pilot
      </span>
    </h1>
    <p>
      There is no summary for this episode.
    </p>
    <img
      alt="Pilot"
      class="sc-fBWQRz EhMCl"
      src="https://static.tvmaze.com/uploads/images/original_untouched/1/4388.jpg"
    />
    <a
      class="sc-aXZVg gvQVpw arrow"
      href="/"
    >
      Return
    </a>
  </article>
</div>
`;
